spec/support/corpus.rb

```diff
@@ -226,6 +245,60 @@ def system(arguments)
         end
       end
 
+      # Mapping of files which we expect to cause errors during mutation generation
+      class ErrorWhitelist
+        class UnnecessaryExpectation < StandardError
+          MESSAGE = 'Expected to encounter %s while mutating "%s"'.freeze
+
+          def initialize(error, path)
+            super(MESSAGE % [error, path])
+          end
+        end
+
+        include Concord.new(:map), Adamantium
+
+        # Assert that we expect to encounter the provided exception for this path
+        #
+        # @param path [Pathname]
+        # @param exception [Exception]
+        #
+        # @raise provided exception if we are not expecting this error
+        #
+        # This method is reraising exceptions but rubocop can't tell
+        # rubocop:disable Style/SignalException
+        #
+        # @return [undefined]
+        def assert_error(path, exception)
+          original_error = exception.cause || exception
+
+          raise exception unless map.fetch(original_error.inspect, []).include?(path)
```

Comment:
Oh heh, I was just looking at this code as if it were part of `Mutant::`. Yeah, I agree with leaving it as-is for now. If `MutantSpec::EMPTY_ARRAY` is ever created then I'd change it, but for now it's cool.